rhino/src/main/java/org/mozilla/javascript/OwnerAwareLambdaSlot.java

```diff
@@ -0,0 +1,88 @@
+package org.mozilla.javascript;
+
+import java.util.IdentityHashMap;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+
+/**
+ * A specialized property accessor using lambda functions, similar to {@link LambdaSlot},
```

Comment:
As far as I understand, the OwnerAwareLambdaSlot is very similar to LambdaSlot. (but code flow is a bit different. see comments)

I made a quick and dirty test, If I replace 
```java
transient Supplier<Object> getter;
transient Consumer<Object> setter;
```
in LambdaSlot with your suggestion
```java
transient Function<Scriptable, Object> getter;
transient BiConsumer<Scriptable, Object> setter;
```
I would need to adjust the existing defineProperty this way.
```java
    public void defineProperty(
            String name, Supplier<Object> getter, Consumer<Object> setter, int attributes) {
        LambdaSlot slot = slotMap.compute(name, 0, ScriptableObject::ensureLambdaSlot);
        slot.setAttributes(attributes);
        slot.getter = (start) -> getter.get();
        slot.setter = (start, value) -> setter.accept(value);
    }
```

What I want to say: Your modified `OwnerAwareLambdaSlot` can cover the whole functionality of `LambdaSlot`

We only require to convert `Supplier<Object>` into a `Function<Scriptable, Object>` and  respectively `Consumer<Object>` into a `BiConsumer<Scriptable, Object>`

The question is, what is more important: Having two classes with nearly the same logic, and save some CPU cycles not compiling an additional lambda or having only one class which gurantees, that code does not diverge. (which already happen)

(!) I have no clear oppinion about this, so maybe wait, if there is an additional feedback

Maybe there is a third option. We could extend the existing `LambdaSlot` this way:

```java
transient Supplier<Object> getter;
transient Function<Scriptable, Object> ownerAwareGetter;
transient Consumer<Object> setter;
transient BiConsumer<Scriptable, Object> ownerAwareSetter;
```
This would only require 2-3 additional null checks, which is IMHO an acceptable tradeoff of additional complexity and CPU cycles

Reply:
Perfectly reasonable point to combine both into single class, although I also don't have strong opinion either way. I wasn't sure what the original intention behind `LambdaSlot` was, so it felt simpler to add another class.

I would welcome more feedback.